vendor/brave-ios/Ledger/BATBraveLedger.mm

```diff
@@ -2003,42 +1644,31 @@ - (void)unblindedTokensReady
   }
 }
 
-- (void)deleteUnblindedTokensForPromotion:(const std::string&)promotion_id callback:(ledger::ResultCallback)callback
-{
-  // TODO please implement
-}
-
-- (void)getTransactionReport:(const ledger::ActivityMonth)month year:(const uint32_t)year callback:(ledger::GetTransactionReportCallback)callback
-{
-  // TODO please implement
-}
-
-- (void)getContributionReport:(const ledger::ActivityMonth)month year:(const uint32_t)year callback:(ledger::GetContributionReportCallback)callback
-{
-  // TODO please implement
-}
-
-- (void)getIncompleteContributions:(ledger::GetIncompleteContributionsCallback)callback
-{
-  // TODO please implement
-}
-
-- (void)getContributionInfo:(const std::string&)contribution_id callback:(ledger::GetContributionInfoCallback)callback
-{
-  // TODO please implement
-}
-
-- (void)updateContributionInfoStepAndCount:(const std::string&)contribution_id step:(const ledger::ContributionStep)step retry_count:(const int32_t)retry_count callback:(ledger::ResultCallback)callback
+- (void)reconcileStampReset
 {
   // TODO please implement
 }
 
-- (void)updateContributionInfoContributedAmount:(const std::string&)contribution_id publisher_key:(const std::string&)publisher_key callback:(ledger::ResultCallback)callback
+- (void)runDBTransaction:(ledger::DBTransactionPtr)transaction
+                callback:(ledger::RunDBTransactionCallback)callback
 {
-  // TODO please implement
+  if (!rewardsDatabase || transaction.get() == nullptr) {
+    auto response = ledger::DBCommandResponse::New();
+    response->status = ledger::DBCommandResponse::Status::ERROR;
+    callback(std::move(response));
+  } else {
+    __block auto transactionClone = transaction->Clone();
+    dispatch_async(self.databaseQueue, ^{
+      __block auto response = ledger::DBCommandResponse::New();
+      rewardsDatabase->RunTransaction(std::move(transactionClone), response.get());
+      dispatch_async(dispatch_get_main_queue(), ^{
+        callback(std::move(response));
+      });
+    });
+  }
 }
 
-- (void)reconcileStampReset
+- (void)pendingContributionSaved:(const ledger::Result)result
 {
   // TODO please implement
```

Comment:
Do these need to be implemented for this PR?

Reply:
@kylehickinson I don't think so as iOS is behind master anyway. But needs to be done when you pull the latest b-c master into iOS

Reply:
we decided on DM to do it in separate PR